File: src/unity/Runtime/Ads/NullAd.cs
using System;
using System.Threading.Tasks;

using Random = UnityEngine.Random;

namespace EE {
    public class NullAd : ObserverManager<AdObserver>, IAd {
        public void Destroy() {
        }

        public bool IsLoaded => false;

        public async Task<bool> Load() {
            await Task.Delay(1000);
            var r = Random.Range(0, 2);
            return r switch {
                0 => true,
                1 => false,
                _ => throw new ArgumentOutOfRangeException()
            };
        }
    }
}

File: src/ts/src/ads/NullAd.ts
import { ObserverManager } from "../core";
import { AdObserver, IAd } from "./IAd";

export class NullAd extends ObserverManager<AdObserver> implements IAd {
    public destroy(): void {
    }

    public get isLoaded(): boolean {
        return false;
    }

    public async load(): Promise<boolean> {
        return false;
    }
}